### PR TITLE
address edge cases in multiple plans

### PR DESCRIPTION
**Pull request by elizabeth-tang**
### Summary 

This pull request addresses some bugs/design decisions found for multiple plans during our testing session

- disallow deleting last plan
- disallow creating two plans with the same name
- character limit for plans
- frontend bugs (colors, borders, etc.)

Remaining TODOs:

- highlight/placeholder text for renaming modal
- some more frontend stuff 

### Test Plan 

image

- Should show warning "A plan with that name already exists." if you try to add a plan with a name that already exists.
- Should not let you delete your last plan
- Should not let you type more than 15 characters.
- Please lmk if you find other issues too!! 

Documentation: https://www.notion.so/Bug-Bash-81ca898208934e9c8a97bcd494adfc34

### CODE REVIEW

File: src/components/Modals/TextInputModal.vue [hunk 1]
```diff
@@ -5,10 +5,13 @@
       <div class="textInput-wrapper">
         <input
           class="textInput-userinput"
-          maxlength="charLimit"
+          maxlength="15"
```
Comment (nidhi-mylavarapu) [inline src/components/Modals/TextInputModal.vue:8]: the 15-character limit works for me! do we also want to require that the plan name cannot be empty?

File: src/components/Modals/EditPlanModal.vue [hunk 2]
```diff
@@ -8,10 +8,12 @@
     @left-button-clicked="closeCurrentModal"
     @right-button-clicked="saveChanges"
     @plan-name="setPlanName"
+    @warn-state="isWarn"
     :isPlanModal="true"
     label="Name"
+    warning="A plan with that name already exists."
```
Comment (nidhi-mylavarapu) [inline src/components/Modals/EditPlanModal.vue:14]: works!

File: src/components/Modals/EditPlanModal.vue [hunk 3]
```diff
@@ -8,10 +8,12 @@
...
   >
-    <button class="editPlan-delete" @click="deletePlan">DELETE PLAN</button>
+    <button class="editPlan-delete" @click="deletePlan" :disabled="!canDelete">DELETE PLAN</button>
```
Comment (nidhi-mylavarapu) [inline src/components/Modals/EditPlanModal.vue:16]: also works - can't delete last plan

**Review by nidhi-mylavarapu — COMMENTED** [review]
overall all the changes mentioned in the PR work and it closes after saving changes! i think we should add a check that the plan name is nonempty

**Comment by andxu282** [comment]
By the way, I think the PR snapshot is failing at the moment because of the outdated node version, if you rebase off `main` it should fix it

**Comment by andxu282** [comment]
Maybe just disable the "Save Changes" button here if there is a dup?
Screenshot 2024-03-04 at 6 48 00 PM

**Pull request merged**